docs/pages/modules/config-plugin-and-native-module-tutorial.mdx

```diff
@@ -0,0 +1,342 @@
+
+export function getApiKey(): string {
+  return ExpoNativeConfigurationModule.getApiKey();
+}
+```
+
+```typescript example/App.tsx
+import * as ExpoNativeConfiguration from 'expo-native-configuration';
+import { Text, View } from 'react-native';
+
+export default function App() {
+  return (
+    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+      <Text>API key: {ExpoNativeConfiguration.getApiKey()}</Text>
+    </View>
+  );
+}
+```
+
+## 3. Run the example project
+
+Now let's run the example project to make sure everything is working. We'll need to start the TypeScript compiler to watch for changes and rebuild the module JavaScript, and separately in another terminal window we will compile and run the example app.
+
+<Terminal
+  cmdCopy="npm run build"
+  cmd={[
+    '# Run this in the root of the project to start the TypeScript compiler',
+    '$ npm run build',
+  ]}
+/>
+
+<Terminal
+  cmdCopy="cd example && npx expo run:ios"
+  cmd={[
+    '$ cd example',
+    '# Run the example app on iOS',
+    '$ npx expo run:ios',
+    '# Run the example app on Android',
+    '$ npx expo run:android',
+  ]}
+/>
+
+We should see a screen with a text saying `API key: api-key`. Now let's develop the plugin to inject our custom API key.
+
+## 4. Creating a new config plugin
+
+With all that set up, we can now start developing our new config plugin. Plugins are synchronous functions that accept an `ExpoConfig` and return a modified `ExpoConfig`. By convention, these functions are prefixed by the word `with` , and in this guide, we will name our plugin `withMyApiKey` — however, you can name it whatever you like, as long as it follows this pattern.
+
+Here is an example of what a basic config plugin function would look like:
+
+```javascript
+const withMyApiKey = config => {
+  return config;
+};
+```
+
+Additionally, you can use `mods`, async functions that modify native project files, such as source code or configuration (plist, xml) files. The `mods` object is different from the rest of the Expo config because it doesn't get serialized after the initial reading, this means you can use it to perform actions *during* code generation.
+
+With that said there are a few considerations that we should follow when writing config plugins:
+
+- Plugins should be synchronous and their return value should be serializable, except for any `mods` that are added.
+- `plugins` are always invoked when the config is read by the `expo/config` method `getConfig`. However, `mods` are only invoked during the "syncing" phase of `npx expo prebuild`.
+
+Although not required, we can use [`expo-module-scripts`](https://www.npmjs.com/package/expo-module-scripts) to make plugin development easier — it provides a recommended default configuration for TypeScript and Jest. Refer to the [config plugins guide](https://github.com/expo/expo/tree/main/packages/expo-module-scripts#-config-plugin) for more information.
```

Comment:
Yeah, I think a callout makes sense here because even though `expo-module-scripts` is included by default when using `create-expo-module`, it is not required for developing expo plugins